src/cryptsetup/cryptsetup-tokens/luks2-tpm2.c

```diff
@@ -58,7 +59,7 @@ int acquire_luks2_key(
         if ((flags & TPM2_FLAGS_USE_PIN) && salt && !pin)
```

Comment:
Still I think this should be also changed to `salt_size > 0`. But OK. Should not be related to the issue explained [here](https://github.com/systemd/systemd/pull/26416#issuecomment-1431264385).